NanoGardener/python/modules/BWEwkSingletReweighter.py

```diff
@@ -555,37 +555,40 @@ def analyze(self, event):
                                                          fourMomenta[0], fourMomenta[1], fourMomenta[2], fourMomenta[3],
                                                          partons, partonIDs,
                                                          mothers, motherIDs)
-            addweight = {}
-            addweight["_I"] = self.mela.weightStoI()
-            #addweight["_I_Honly"] = self.mela.weightStoI_H()
-            #addweight["_I_Bonly"] = self.mela.weightStoI_B()
-            #addweight["_I_HB"] = self.mela.weightStoI_HB()
-            addweight["_B"] = self.mela.weightStoB()
-            #addweight["_H"] = self.mela.weightStoH()
-
-            for appendix in self.branches:
-              if math.isnan(addweight[appendix]) or math.isinf(addweight[appendix]): #dirty protection for occasional failures
-                self.NANinfo
-                addweight[appendix]=0.
-              weights[name+appendix] = weights[name]*addweight[appendix]
-
-            for appendix in [""]+self.branches:
-              self.out.fillBranch(name+appendix, weights[name+appendix])
-
+            #addweight = {}
+            weights[name+"_I"] = math.sqrt(weights[name])*self.mela.weightStoI()
+            weights[name+"_I_Honly"] = math.sqrt(weights[name])*self.mela.weightStoI_H()
+            weights[name+"_I_Bonly"] = math.sqrt(weights[name])*self.mela.weightStoI_B()
+            # now reset to the SM high mass signal and do not touch the normalization again
+            self.mela.setMelaHiggsMassWidth(self.mH, self.gsm)
+            weights[name+"_I_HB"] = self.mela.weightStoI_HB()
+            # reset to SM width before reweighting to H and B
+            self.mela.setMelaHiggsMassWidth(self.mH, self.gsm)
+            weights[name+"_B"] = self.mela.weightStoB()
+            weights[name+"_H"] = self.mela.weightStoH()
+ 
+            for key in weights.keys():
+              if math.isnan(weights[key]) or math.isinf(weights[key]): #dirty protection for occasional failure
+                weights[key] = 0.
+              self.out.fillBranch(key, weights[key])
 
         ########## For Relative Width & Fixed Width model
+        gprime = self.gsm
         for relw in self.relw_list:
-          if relw < 1.0: # Relative width
-            self.gsm = relw * self.mH
+          if relw == "orig":
+            # SM width
+            pass  
+          elif relw < 1.0: # Relative width
+            gprime = relw * self.mH
           else: # Fixed width
-            self.gsm = relw
-          if self.gsm == 0: self.gsm = 0.001 * self.mH # Is actually 0.1% relative width
+            gprime = relw
+          if gprime == 0: self.gsm = 0.001 * self.mH # Is actually 0.1% relative width
```

Comment:
`gprime = 0.001 * self.mH`

Reply:
strange, I don't remember why I changed this

Reply:
It's because you need to keep `self.gsm` to put back to SM width later, right? My comment here is just that the `self.gsm` here needs to be changed to `gprime`